html/js/product-preferences.js

```diff
@@ -84,6 +85,39 @@ function display_selected_preferences (target_selected_summary, product_preferen
 }
 
 
+function generate_preferences_switch_button(preferences_text, checkbox_id) {
+
+	var checked = '';
+	if (use_user_product_preferences_for_ranking) {
+		checked = " checked";
+	}	
+
+	var html = '<fieldset class="switch round success" tabindex="0" id="' + checkbox_id +'_switch" style="float:left;margin:0;margin-right:.5rem;padding-top:0.1rem;">'
+	+ '<input class="preferences_checkboxes" id="' + checkbox_id + '" type="checkbox"' + checked + '>'
+	+ '<label for="' + checkbox_id +'"></label></fieldset>'
+	+ '<label for="' + checkbox_id +'" style="float:left;margin-right:1em;padding-top:4px">' + preferences_text + '</label>';
```

Comment:
One day you will embrace javascript template strings :-P

Reply:
I'll do it in another PR :)